gadgets/trace_dns/program.bpf.c

```diff
@@ -340,23 +342,22 @@ int ig_trace_dns(struct __sk_buff *skb)
 	gadget_process_populate_from_socket(skb_val, &event->proc);
 
 	if (paths && skb_val != NULL) {
-		unsigned int cwd_len = min(bpf_core_field_size(skb_val->cwd),
-					   TRACE_DNS_PATH_MAX);
-		unsigned int exepath_len =
-			min(bpf_core_field_size(skb_val->exepath),
-			    TRACE_DNS_PATH_MAX);
 		bool cwd_exists = bpf_core_field_exists(skb_val->cwd);
 		bool exepath_exists = bpf_core_field_exists(skb_val->exepath);
 		bool probe_read_kernel_str_exists = bpf_core_enum_value_exists(
 			enum bpf_func_id, BPF_FUNC_probe_read_kernel_str);
 
 		if (probe_read_kernel_str_exists) {
 			if (cwd_exists) {
+				unsigned int cwd_len =
+					bpf_core_field_size(event->cwd);
 				bpf_probe_read_kernel_str(&event->cwd, cwd_len,
 							  skb_val->cwd);
 			}
 
 			if (exepath_exists) {
+				unsigned int exepath_len =
+					bpf_core_field_size(event->exepath);
 				bpf_probe_read_kernel_str(&event->exepath,
 							  exepath_len,
 							  skb_val->exepath);
```

Comment:
How does this ensure that the size of `event->exepath` is not bigger than `skb_val->exepath`?

Reply:
Good catch. I added the missing check there.